### PR TITLE
v1

### PR DESCRIPTION
**Pull request by michaelkerry**

### CODE REVIEW

File: pyslack/notify.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+    COLOR_SUCCESS = "#00b600"
+    COLOR_INFO = "#7799b9"
+    COLOR_ERROR = "#ef0000"
+    COLOR_WARNING = "#ffce03"
+
+    ICON_SUCCESS = "white_check_mark"
+    ICON_INFO = "information_source"
+    ICON_ERROR = "x"
+    ICON_WARNING = "warning"
```
Comment (ccurreri) [inline pyslack/notify.py]: Consider an `Enum` for these:

```
from enum import Enum

class Color(Enum):
    SUCCESS = "#00b600"
    INFO = "#7799b9"
    ERROR = "#ef0000"
    WARNING = "#ffce03"

class Icon(Enum):
    SUCCESS = "white_check_mark"
    INFO = "information_source"
    ERROR = "x"
    WARNING = "warning"

Color.SUCCESS.value()
Icon.ERROR.value()
```

This gives you a few things:
1. Your IDE can now lint it more easily and display you available options
2. You can't accidentally use a color or icon that isn't implemented
3. Implementing a new color or icon is as simple as adding it to the enum (no need to change logic or variables)

**Review by robruma — APPROVED** [review]

File: pyslack/notify.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        :param title:
+        :param message:
+        :param link:
+        :param link_title:
+        :return:
+        """
+        fallback = self.setup_fallback(self.ICON_SUCCESS, title, link, link_title)
+        attachments = self.setup_attachments(self.COLOR_SUCCESS, fallback, message)
+        requests.post(self.slack_url,
+                      data={"payload": self.slack_payload_with_attachments(attachments)})
```
Comment (ccurreri) [inline pyslack/notify.py]: This, and the following methods, can be abstracted:

    def send_msg(self, title: str, msg_color: Color, msg_icon: Icon, message: str, link: str = None, link_title: str = None):
        fallback = self.setup_fallback(msg_icon.value(), title, link, link_title)
        attachments = self.setup_attachments(msg_color.value(), fallback, message)
        requests.post(self.slack_url,
                      data={"payload": self.slack_payload_with_attachments(attachments)})

    def success(self, title: str, message: str = "Success!", link: str = None, link_title: str = None):
        send_msg(title, Color.SUCCESS, Icon.SUCCESS, message, link, link_title)

    def info(self, title: str, message: str, link: str = None, link_title: str = None):
        send_msg(title, Color.INFO, Icon.INFO, message, link, link_title)
    
    . . .

**Review by ccurreri — APPROVED** [review]

File: pyslack/notify.py (hunk 2 above)
Comment (michaelkerry) [reply pyslack/notify.py]: good point! updated

File: pyslack/notify.py (hunk 1 above)
Comment (michaelkerry) [reply pyslack/notify.py]: good points! updated

File: README.md [hunk 3]
```diff
@@ -1 +1,33 @@
...
+### Instantiation Notes
+    1. <unique_id_for_slack_webhook> must be supplied by slack, and will be appended to "https://hooks.slack.com/services/"
+    1. <notification_username> can be any string, not necessarily an existing slack user
+    1. <emoji> must be a valid slack emoji name - do not include any ':' as it will automatically be wrapped ":<emoji>:"
+    
+### Parameters
+    1. title -> REQUIRED
+    1. message -> REQUIRED ( defaults to 'Success!' for .success() )
+    1. link -> OPTIONAL ( must be a fully formed URL, e.g., https://www.example.com )
+    1. link_title -> OPTIONAL ( if link parameter is supplied but not link_title, link_title defaults to 'link' )
```
Comment (jazahn) [inline README.md]: The numerated list isn't rendering in github: https://github.com/huit/pyslack/blob/development/README.md

File: README.md [hunk 4]
```diff
@@ -1 +1,33 @@
...
+    
+    ns.success("SUCCESS TITLE") # only title required for success message
+    ns.info(title="INFO TITLE", message="informational message")
+    ns.warning(title="WARNING TITLE", message="warning message")
+    ns.error(title="ERROR TITLE", message="error message", link="https://www.example.com", link_title="Example.com")
+
+### Instantiation Notes
+    1. <unique_id_for_slack_webhook> must be supplied by slack, and will be appended to "https://hooks.slack.com/services/"
+    1. <notification_username> can be any string, not necessarily an existing slack user
+    1. <emoji> must be a valid slack emoji name - do not include any ':' as it will automatically be wrapped ":<emoji>:"
```
Comment (jazahn) [inline README.md]: I'm thinking a heading for getting the slack webhook set up would be useful. I'm not sure if a direct link like https://harvard-huit.slack.com/apps/A0F7XDUAZ-incoming-webhooks?next_id=930064566807&direction=forwards would be enough -- I think you'd probably want a 
 - in slack, click "customize slack"
 - brings you to a webpage, click "configure apps"
 - find 'incoming webhooks' 
 - click 'request configuration'

**Review by jazahn — APPROVED** [review]
Just a couple README suggestions. Nice work!

File: README.md (hunk 3 above)
Comment (michaelkerry) [reply README.md]: yes, that was bc the lines were indented, so the 'formatting' was preserved. fixed

File: README.md (hunk 4 above)
Comment (michaelkerry) [reply README.md]: I'm reluctant to put a link to harvard-huit slack... is the other link, more of a larger overview of how to set up webhooks, good enough?

**Review by jazahn — APPROVED** [review]
LGTM

**Pull request merged**